sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1574,7 +1579,7 @@ class Analyzer(
                     s"its class is ${other.getClass.getCanonicalName}, which is not a generator.")
               }
             }
-          case u @ UnresolvedFunction(funcId, children, isDistinct) =>
+          case u @ UnresolvedFunction(funcId, children, isDistinct, filter) =>
```

Comment:
Could you throw an analysis exception if `filter` given  in non-aggregation functions?

Reply:
@maropu Thanks for your remind. I will add it.